src/main/java/org/dcsa/ebl/service/impl/ShippingInstructionServiceImpl.java

```diff
@@ -292,6 +299,25 @@
       }
     }
 
+    // Check if documentParties is not empty and that partyFunction on DocumentParty set properly
+    // for a non-electronic shipping instruction
+    if (Objects.nonNull(shippingInstructionTO.getIsElectronic())
+        && shippingInstructionTO.getIsElectronic()) {
+      if (Objects.isNull(shippingInstructionTO.getDocumentParties())
+          || shippingInstructionTO.getDocumentParties().isEmpty()) {
+        validationErrors.add(
+            "A documentParty with partyFunction=EBL is required for electronic shipping instructions.");
+      }
+
+      if (Objects.nonNull(shippingInstructionTO.getDocumentParties())
+          && shippingInstructionTO.getDocumentParties().stream()
+                  .filter(x -> x.getPartyFunction().equals(PartyFunction.EBL))
+                  .count()
+              != 1) {
+        validationErrors.add("Only 1 EBL solution provider can be specified in DocumentParties");
```

Comment:
If number of DocumentParties with EBL = 0 then the error message is wrong... Please make 2 error messages depending on 0 EBL specified and multiple EBL specified

Reply:
Done